src/hooks/useEarthEngineLayersStore.js

```diff
@@ -0,0 +1,103 @@
+    const getLayers = useCallback(() => {
+        setLoading(true)
+        engine
+            .query({ dataStore: { resource: 'dataStore' } })
+            .then(({ dataStore }) => {
+                if (dataStore.includes(MAPS_APP_NAMESPACE)) {
+                    // Fetch layers if namespace/keys exists in data store
+                    engine
+                        .query({ layers: { resource } })
+                        .then(({ layers }) => {
+                            setAddedLayers(layers)
+                            setLoading(false)
+                        })
+                } else {
+                    // Create namespace/keys if missing in data store
+                    engine
+                        .mutate({
+                            resource,
+                            type: 'create',
+                            data: [],
+                        })
+                        .then((response) => {
+                            if (response.httpStatusCode === 201) {
+                                setLoading(false)
+                            } else {
+                                setError(response)
+                            }
+                        })
+                        .catch(setError)
+                }
+            })
+    }, [engine])
+
+    // Add layer id to data store
+    const addLayer = useCallback(
+        (layerId) => {
+            if (!addedLayers.includes(layerId)) {
+                engine
+                    .mutate({
+                        resource,
+                        type: 'update',
+                        data: [...addedLayers, layerId],
+                    })
+                    .then((response) => {
+                        if (response.httpStatusCode === 200) {
+                            getLayers()
+                        } else {
+                            setError(response)
+                        }
+                    })
+                    .catch(setError)
+            }
+        },
+        [engine, addedLayers, getLayers]
+    )
+
+    // Remove layer id from data store
+    const removeLayer = useCallback(
+        (layerId) => {
+            if (addedLayers.includes(layerId)) {
+                engine
+                    .mutate({
+                        resource,
+                        type: 'update',
+                        data: addedLayers.filter((l) => l !== layerId),
+                    })
+                    .then((response) => {
+                        if (response.httpStatusCode === 200) {
+                            getLayers()
+                        } else {
+                            setError(response)
+                        }
+                    })
+                    .catch(setError)
+            }
+        },
+        [engine, addedLayers, getLayers]
+    )
+
+    useEffect(() => {
+        getLayers()
+    }, [engine, getLayers])
+
+    return { addedLayers, loading, addLayer, removeLayer, error }
```

Comment:
error is returned but I can't see that it is used anywhere. Maybe there should be some error handling, like showing a notice in the case that an add/remove operation failed.